File: frontend/src/types/agent.ts
export interface SandboxConfig {
  memory?: string
  cpu?: number
  timeout?: number
}

export interface AgentResponse {
  id: string
  user_id: string
  name: string
  description: string | null
  icon: string | null
  instructions: string | null
  model_config_json: Record<string, unknown> | null
  tool_categories: string[] | null
  suggested_prompts: string[] | null
  kb_ids: string[] | null
  connector_ids: string[] | null
  mcp_server_ids: string[] | null
  grounding_config: Record<string, unknown> | null
  sandbox_config: SandboxConfig | null
  execution_mode: "react" | "dag" | "auto"
  status: string
  is_active: boolean
  visibility?: string // "personal" | "org" | "global"
  org_id?: string | null
  is_builder?: boolean
  compact_instructions?: string | null
  forked_from: string | null
  publish_status: string | null
  reviewed_by: string | null
  reviewed_at: string | null
  review_note: string | null
  published_at: string | null
  created_at: string
  updated_at: string | null
}

export interface AgentCreate {
  name: string
  description?: string | null
  icon?: string | null
  instructions?: string | null
  model_config_json?: Record<string, unknown>
  tool_categories?: string[]
  suggested_prompts?: string[]
  kb_ids?: string[]
  connector_ids?: string[]
  mcp_server_ids?: string[]
  grounding_config?: Record<string, unknown>
  sandbox_config?: SandboxConfig
  execution_mode?: "react" | "dag" | "auto"
  compact_instructions?: string | null
}

export interface AgentUpdate {
  name?: string
  description?: string | null
  icon?: string | null
  instructions?: string | null
  model_config_json?: Record<string, unknown>
  tool_categories?: string[]
  suggested_prompts?: string[]
  kb_ids?: string[]
  connector_ids?: string[]
  mcp_server_ids?: string[]
  grounding_config?: Record<string, unknown>
  sandbox_config?: SandboxConfig
  execution_mode?: "react" | "dag" | "auto"
  compact_instructions?: string | null
}

export interface AgentTemplate {
  id: string
  name: string
  description: string
  icon?: string | null
  category: string
  blueprint: Record<string, unknown>
}

export interface AICreateAgentResult {
  agent: AgentResponse
  message: string
  message_key?: string
  message_args?: Record<string, unknown>
}

export interface AIRefineAgentResult {
  agent: AgentResponse
  modified_fields: string[]
  message: string
  message_key?: string
  message_args?: Record<string, unknown>
}

File: src/fim_one/web/schemas/agent.py
"""Agent request/response schemas."""

from __future__ import annotations

from typing import Literal

from pydantic import BaseModel, Field


class AgentCreate(BaseModel):
    name: str = Field(min_length=1, max_length=200)
    icon: str | None = None
    description: str | None = None
    instructions: str | None = None
    model_config_json: dict | None = None
    tool_categories: list[str] | None = None
    suggested_prompts: list[str] | None = None
    kb_ids: list[str] | None = None
    connector_ids: list[str] | None = None
    grounding_config: dict | None = None
    sandbox_config: dict | None = None
    execution_mode: Literal["react", "dag", "auto"] = "auto"


class AgentUpdate(BaseModel):
    name: str | None = None
    icon: str | None = None
    description: str | None = None
    instructions: str | None = None
    model_config_json: dict | None = None
    tool_categories: list[str] | None = None
    suggested_prompts: list[str] | None = None
    kb_ids: list[str] | None = None
    connector_ids: list[str] | None = None
    grounding_config: dict | None = None
    sandbox_config: dict | None = None
    execution_mode: Literal["react", "dag", "auto"] | None = None


class AgentResponse(BaseModel):
    id: str
    user_id: str
    name: str
    icon: str | None
    description: str | None
    instructions: str | None
    model_config_json: dict | None
    tool_categories: list[str] | None
    suggested_prompts: list[str] | None
    kb_ids: list[str] | None
    connector_ids: list[str] | None
    grounding_config: dict | None
    sandbox_config: dict | None
    execution_mode: str
    status: str
    published_at: str | None
    is_builder: bool = False
    discoverable: bool = False
    sub_agent_ids: list[str] | None = None
    visibility: str = "personal"
    org_id: str | None = None
    publish_status: str | None = None
    reviewed_by: str | None = None
    reviewed_at: str | None = None
    review_note: str | None = None
    created_at: str
    updated_at: str | None


class AICreateAgentRequest(BaseModel):
    instruction: str = Field(min_length=1, max_length=5000)


class AIRefineAgentRequest(BaseModel):
    instruction: str = Field(min_length=1, max_length=2000)
    history: list[dict] = Field(default_factory=list)


class AICreateAgentResult(BaseModel):
    agent: AgentResponse
    message: str = ""
    message_key: str = ""
    message_args: dict = {}


class AIRefineAgentResult(BaseModel):
    agent: AgentResponse
    modified_fields: list[str] = []
    message: str = ""
    message_key: str = ""
    message_args: dict = {}
